teams/views.py

```diff
@@ -213,66 +219,106 @@ def rename_team(request, team_id):
 
 @login_required
 @has_access_to_team()
-def create_group_im(request, team_id):
-    """ Creates a new group IM in slack """
+def create_private_channel(request, team_id):
+    """ Creates a new Private Slack Channel in slack """
+    slack_site_settings = SlackSiteSettings.objects.first()
     if request.method != 'POST':
         messages.error(request,
                        'You cannot access this page in this way.')
         return redirect(reverse('view_team', kwargs={'team_id': team_id}))
 
     team = get_object_or_404(HackTeam, id=team_id)
+    non_participant_superuser = (
+        (request.user.is_superuser or request.user.is_staff)
+        and not request.user.is_participant(team.hackathon))
+
     if not (request.user in team.participants.all()
-            or request.user == team.mentor):
+            or request.user == team.mentor or non_participant_superuser):
         messages.error(request,
-                       ('You do not have access to create a Slack IM group '
+                       ('You do not have access to create a Private Slack Channel '
                         'for this team.'))
         return redirect(reverse('view_team', kwargs={'team_id': team_id}))
 
-    if not (settings.SLACK_ENABLED or settings.SLACK_BOT_TOKEN
-            or settings.SLACK_WORKSPACE):
+    if (not (settings.SLACK_ENABLED or settings.SLACK_BOT_TOKEN
+            or settings.SLACK_WORKSPACE) or slack_site_settings.communication_channel_type != 'slack_private_channel'):
         messages.error(request, 'This feature is currently not enabled.')
         return redirect(reverse('view_team', kwargs={'team_id': team_id}))
 
+    # Create new channel
+    date_str = datetime.now().strftime('%y%m')
+    team_name = re.sub('[^A-Za-z0-9]+', '', team.display_name.lower())
+    channel_name = f'{date_str}-hackathon-{team_name}'
```

Comment:
brilliant! this was going to be my first question... what're the private channel names and convention... nice! 👍🏻

Reply:
Thanks :+1: